### PR TITLE
Base1.4

### PR DESCRIPTION
**Pull request by redheadedrod**
This IS the last of my current code base. You know have all of my current code baring any errors during merge that I missed. 

A bunch of new stuff was added in this one.

### CODE REVIEW

File: envi/archs/arm/disasm.py [hunk 1]
```diff
@@ -1012,17 +1022,25 @@ def p_media_pack_sat_rev_extend(opval, va):
...
         Rd = (opval>>12) & 0xf
         rot = (opval>>10) & 3
         Rm = opval & 0xf
-        
-        olist = (
-            ArmRegOper(Rd, va=va),
-            ArmRegOper(Rn, va=va),
-            ArmRegShiftImmOper(Rm, S_ROR, rot, va),
-        )
+        print idx, Rn
```
Comment (atlas0fd00m) [inline envi/archs/arm/disasm.py]: seriously? :)

File: envi/archs/arm/disasm.py [hunk 2]
```diff
@@ -1881,6 +1899,59 @@ def repr(self, op):
             retval.append(shift_names[self.shtype])
         return " ".join(retval)
 
+class ArmRegShiftXtndOper(ArmOperand):
```
Comment (atlas0fd00m) [inline envi/archs/arm/disasm.py]: how does this differ from ArmRegShiftImmOper?

File: envi/archs/arm/disasm.py (hunk 1 above)
Comment (redheadedrod) [inline envi/archs/arm/disasm.py]: Missed one....

File: envi/archs/arm/disasm.py (hunk 2 above)
Comment (redheadedrod) [inline envi/archs/arm/disasm.py]: See top comment for class.
ShiftImmOper has exact value to shift in the instruction. (Example such as ROR A8.8.149)
ShiftXtndOper can only rotate by 0, 8, 16 or 32 depending on bits 10 and 11 which are the selector bits. The Xtend name comes from the fact these are all sign extend commands.
If you look at SXTAB (A8.8.230 in ARMV7 reference) it shows an example.

**Pull request merged**